### PR TITLE
Adding github actions for protbufs

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Adds a chunk of the github actions files.
will add the buf ones later as those tools fail when there is a lack of `proto` files. See https://github.com/nick-mobilecoin/enclave-proto/pull/6 and https://github.com/nick-mobilecoin/enclave-proto/pull/7 for what's coming with those actions

### CODE REVIEW

File: .github/dependabot.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version: 2
```
Comment (nick-mobilecoin) [inline .github/dependabot.yaml:1]: Note, standardized on `yaml` since that's the suffix the yaml spec suggests.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - uses: xt0rted/markdownlint-problem-matcher@v2
+      - uses: DavidAnson/markdownlint-cli2-action@v9
+        with:
+          globs: "**/*.md"
+
+  notify:
+    runs-on: ubuntu-latest
+    if: ${{always()}} && ${{failure()}} && ${{ github.event_name }} == 'push'
```
Comment (nick-mobilecoin) [inline .github/workflows/ci.yaml:21]: I haven't tested this change out, but I noticed this isn't actually working in the sgx repo.
The sgx repo has
```yaml
if: failure() && ${{ github.event_name }} == 'push'
```
it maybe that the `always` part is unecessary and the if condition is enough to allow the needs to follow through when previous steps fail.  I also think the github context variables must have changed `github.repository_url` doesn't seem to exist in docs, neither does `github.pusher` and `github.url`

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/probufs/1" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/probufs/1?utm_source=stack-comment).

**Review by jcape — APPROVED** [review]

**Review by samdealy — APPROVED** [review]
LGTM

**Pull request merged**